session/session.go

```diff
@@ -0,0 +1,34 @@
+package session
+
+import (
+	"fmt"
+	"time"
+
+	"launchpad.net/gozk"
+)
+
+var ErrNotConnected = fmt.Errorf("Unable to connect to ZooKeeper.")
+
+type ZkSession struct {
+	Connection *zookeeper.Conn
+	Events     <-chan zookeeper.Event
+}
+
+func NewZkSession(servers string) (*ZkSession, error) {
+	zkconn, zkevents, err := zookeeper.Dial(servers, 3*time.Second)
+
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case event := <-zkevents:
+		if event.State != zookeeper.STATE_CONNECTED {
```

Comment:
Also sort of important to think about what happens if our connection to ZK drops -- conceptually we no longer hold the lock. Should we stop processing our locked operation?